calls/Space.php

```diff
@@ -161,4 +169,62 @@ public static function leaveSpace($event)
             );
         }
     }
+
+    /**
+     * Enable ethereum integration for already existing spaces
+     *
+     * @param $event
+     * @throws GuzzleException
+     * @throws HttpException
+     * @throws Exception
+     */
+    public function enable($event)
+    {
+        $space = $event->sender;
+
+        if (!$space instanceof BaseSpace) {
+            return;
+        }
+
+        $spaceDefaultAccount = Account::findOne([
+            'space_id' => $space->id,
+            'account_type' => Account::TYPE_DEFAULT
+        ]);
+
+        if (!$spaceDefaultAccount) {
+            Utils::createDefaultAccount($space);
+        } else {
+            Wallet::createWallet(new Event(['sender' => $spaceDefaultAccount]));
+            Dao::createDao($event);
+        }
+
+        // add space members to created dao
+        foreach ($space->getMemberships()->all() as $memberShip) {
+
+            $memberShipEvent = new MemberEvent([
+                'space' => $space, 'user' => $memberShip->getUser()->one()
+            ]);
+
+            self::addMember($memberShipEvent);
+        }
+
+        $asset = Utils::issueSpaceAsset($space);
+
+        $transactions = Transaction::findAll([
+            'asset_id' => $asset->id,
+        ]);
+
+        foreach ($transactions as $transaction) {
+            $transactionEvent = new Event(['sender' => $transaction]);
+            if ($transaction->transaction_type == Transaction::TRANSACTION_TYPE_ISSUE) {
+                // mint coins for each issue transaction of the space
+                Coin::mintCoin($transactionEvent);
+            } else {
+                //transfer coins for each coin holder
+                Coin::transferCoin($transactionEvent);
+            }
```

Comment:
I would not do all transfers but only call a `mintCoin` for each owner of an asset with the user's current balance? 
this is probably more correct but could lead to many ethereum transactions.
but as probably do not have that many transfers it might be fine?

Reply:
I this even with transfer will be fine , because as you said currently we don't have many transfers to do.